frontend/src/components/gradient.ts

```diff
@@ -39,12 +39,20 @@ export const gradientOptions = (
   const func = d3[key];
   const label = key.replace("interpolate", "");
   const id = label.toLowerCase();
-  const colors = range(0, 1.01, 0.1).map(func);
+  const colors = range(0, 1, 0.1).concat([1]).map(func);
```

Comment:
I assume you're adding 1 to make the range function inclusive of the endpoint. I wonder if this would be good as a helper function just so someone doesn't have to think about why you're adding [1] to the end.